apps/condo/domains/resident/tasks/sendBillingReceiptsAddedNotificationForOrganizationContextTask.js

```diff
@@ -0,0 +1,246 @@
+    // TODO(DOMA-3376): Detect locale by resident locale instead of organization country.
+    const country = get(resident, 'residentOrganization.country', conf.DEFAULT_LOCALE)
+    const locale = get(COUNTRIES, country).locale
+    const notificationKey = makeMessageKey(receipt.id, receipt.period, receipt.account.number, receipt.category.id, resident.id)
+    const toPayValue = parseFloat(receipt.toPay)
+    const toPay = isNaN(toPayValue) ? null : toPayValue
+    const toPayChargeValue = parseFloat(get(receipt, 'toPayDetails.charge'))
+    const toPayCharge = isNaN(toPayChargeValue) ? null : toPayChargeValue
+    const category = getLocalized(locale, receipt.category.nameNonLocalized)
+    const currencyCode = get(receipt, 'context.integration.currencyCode') || DEFAULT_CURRENCY_CODE
+
+    // Temporarily disabled checking toPayCharge value (it's temporarily not used by now)
+    // if (isNull(toPay) || isNull(toPayCharge)) return 0
+    // Disabled notifications of BILLING_RECEIPT_ADDED_WITH_NO_DEBT_TYPE type due to DOMA-6589
+    if (isNull(toPay) || toPay <= 0) return 0
+
+    const { messageType, debt } = getMessageTypeAndDebt(toPay, toPayCharge)
+
+    const data = {
+        residentId: resident.id,
+        userId: resident.user,
+        url: `${conf.SERVER_URL}/billing/receipts/${receipt.id}`,
+        billingReceiptId: receipt.id,
+        billingAccountNumber: receipt.account.number,
+        billingPropertyId: receipt.property,
+        period: receipt.period,
+        category,
+        toPay: debt,
+        currencySymbol: CURRENCY_SYMBOLS[currencyCode] || currencyCode,
+    }
+
+    const messageData = {
+        lang: locale,
+        to: { user: { id: resident.user } },
+        type: messageType,
+        meta: { dv: 1, data },
+        sender: { dv: 1, fingerprint: 'send-billing-receipts-added-notifications' },
+        uniqKey: notificationKey,
+        organization: { id: resident.organization },
+    }
+
+    const { isDuplicateMessage } = await sendMessage(context, messageData)
+
+    return isDuplicateMessage ? 0 : 1
+}
+
+/**
+ * Prepares data for sendMessage to resident on available billing receipt, then tries to send the message
+ * @param {BillingIntegrationOrganizationContext} context
+ * @param {string} lastSyncDate
+ * @returns {Promise<void>}
+ */
+async function sendBillingReceiptsAddedNotificationForOrganizationContext (context, lastSyncDate) {
+    const contextId = get(context, 'id')
+    if (!contextId) throw new Error(`Invalid BillingIntegrationOrganizationContext, cannot get context.id. Context: ${context}`)
+
+    const { keystone } = getSchemaCtx('Message')
+    const redisClient = getRedisClient()
+    const { thisMonthStart } = getStartDates()
+    const receiptCreatedAfter = lastSyncDate || thisMonthStart
+
+    const receipts = await fetchReceipts(contextId, receiptCreatedAfter)
+    logger.info({ msg: `Found ${receipts.length} receipts`, contextId })
```

Comment:
It's better to set a static string as value for `msg`. And move uncommon fields to unindexed `data` field